examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,274 @@
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# Soon we realize that we cannot compare different coefficients since we did
+# not scale the data before the fit, features having different value ranges.
+# For instance, the "AGE" coefficient is expressed in $/hours/leaving years
+# while the "EDUCATION" is expressed in $/hours/years of education.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std().plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# We should then normalize the coefficients by the standard deviation and we
+# will be able to compare them.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_ *
+    X_train_preprocessed.std(),
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The plot above tells us that an increase of the "AGE" will induce a decrease
+# of the "WAGE" when all other features remain constant. Also an increase of
+# the "EXPERIENCE" will induce an increase of the "WAGE" when all other
+# features remain constant.
+#
+# The first interpretation might look counter-intuitive at first, if one
+# relates the relationship between "AGE" and "WAGE" as a marginal link.
+# However, as previously mentioned, a linear model computes a conditional
+# link between "AGE" and "WAGE" given all other features.
+# Therefore, one should also interpret that for a given experience and all
+# other features constant as well, a younger person would have an higher
+# wage.
+#
+# Checking the coefficient stability
+# ..................................
+#
+# The stability of the coefficients is a guarantee of the robustness of the
+# model. We can check the coefficient stability through cross-validation.
+
+from sklearn.model_selection import cross_validate
+from sklearn.model_selection import RepeatedKFold
+
+cv_model = cross_validate(
+    model, X, y, cv=RepeatedKFold(n_splits=5, n_repeats=5),
+    return_estimator=True, n_jobs=-1
+)
+coefs = pd.DataFrame(
+    [est.named_steps['transformedtargetregressor'].regressor_.coef_ *
+     X_train_preprocessed.std()
+     for est in cv_model['estimator']],
+    columns=feature_names
+)
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefs, orient='h', color='k', alpha=0.5)
+sns.boxplot(data=coefs, orient='h', color='blue')
```

Comment:
I am picky but the blue is too much aggresive, meaning that we should reduce the alpha. We don't see the median value for instance.

Reply:
cyan is better?